plugins/snowflake/dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -46,20 +47,44 @@
 
   {% if existing_relation is none %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
+  
   {% elif existing_relation.is_view %}
     {#-- Can't overwrite a view with a table - we must drop --#}
     {{ log("Dropping relation " ~ target_relation ~ " because it is a view and this model is a table.") }}
     {% do adapter.drop_relation(existing_relation) %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
+  
   {% elif full_refresh_mode %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
+  
   {% else %}
     {% do run_query(create_table_as(True, tmp_relation, sql)) %}
-    {% do adapter.expand_target_column_types(
+    
```

Comment:
As with the default materialization, let's consolidate this logic to:
1. Always expand target column types first thing
2. Check for schema changes if and only if `on_schema_change != 'ignore'`. If there are schema changes, process them, endif.
3. Grab `dest_columns` and set `build_sql`

Reply:
this is done for snowflake and porting to the other adapters